cdap-api/src/main/java/io/cdap/cdap/api/app/ApplicationConfigurer.java

```diff
@@ -143,4 +143,15 @@ default RuntimeConfigurer getRuntimeConfigurer() {
   default String getDeployedNamespace() {
     throw new UnsupportedOperationException("Getting deployed namespace is not supported");
   }
+
+  /**
+   * Return the application Specification if it was deployed before. This is useful for upgrade
```

Comment:
I think this will be useful for even execution case now as we will be deprecating `getDeployedApplicationSpec()` from `RuntimeConfigurer`.
Please add deprecation annotation to corresponding fields/methods in `AppDeploymentRuntimeInfo` and `RuntimeConfigurer`